Calculator new features

mpozdeev opened the pull request.

mpozdeev committed:
Исправлена ошибка в Validator_1

Исправлена ошибка в логике работы Validator_1. Ранее всегда выводил false.

mpozdeev committed:
Добавлен логгер

Добавлен файл настройки log4j2.xml;
добавлено логирование в классы: Application, Validator_1, CalculateImp_1

mpozdeev committed:
Много изменений

Переделана структура проекта
Изменены имена классов и интерфейсов
Добавлен класс перевода строки в постфиксную запись
Изменён валидатор - добавлена проверка на корректное окончание выражения

mpozdeev committed:
Добавлен метод вычисления по RPN записи

Добавлена логика записи отрицательных чисел из мат выражения в rpn.

mpozdeev committed:
Добавлены юнит тесты

IliaIsakhin commented on src/main/java/com/maksim/service/calculate/MyCalculator.java:
```diff
@@ -0,0 +1,89 @@
+        LinkedList<String> strPostfix = PostfixRecord.strToPostfix(mathExpression);
+
+//        производим вычисление значения выражения из обратной польской нотации:
+        LinkedList<String> outPostfixString = new LinkedList<>();
```
Не юзается

mpozdeev replied:
Удалил

mpozdeev committed:
Удалены лишние комментарии и закоммиченный код

IliaIsakhin commented on src/main/java/com/maksim/Application.java:
```diff
@@ -1,19 +1,29 @@
-import com.maksim.calculated.CalculateImp_1;
+import com.maksim.service.calculate.Calculator;
+import com.maksim.service.calculate.MyCalculator;
+import org.apache.logging.log4j.LogManager;
```
slf4j

mpozdeev replied:
Заменил

IliaIsakhin commented on src/main/java/com/maksim/service/calculate/MyCalculator.java:
```diff
@@ -0,0 +1,83 @@
+        logger.info("Method: calculate() - init");
+        double result;
+
+        MyValidator validator_My_ = new MyValidator();
```
camel case

mpozdeev replied:
Исправил

IliaIsakhin commented on src/main/java/com/maksim/service/Operations.java:
```diff
@@ -0,0 +1,44 @@
+    private static final Map<String, Integer> MATH_OPERATIONS = new HashMap<>();
+
+    static {
+        MATH_OPERATIONS.put("(", 5);
```
Операторы можно вынести в константы

IliaIsakhin replied:
Или даже в енам

IliaIsakhin commented on src/main/resources/Any.txt:
```diff
@@ -0,0 +1,78 @@
+public class Experimental {
+    public static void main(String[] args) throws Exception {
```
Это зачем?

mpozdeev replied:
Это был временный класс, в котором запускал и проверял работу методов по отдельности.

IliaIsakhin commented on src/test/java/com/maksim/service/OperationsTest.java:
```diff
@@ -0,0 +1,74 @@
+import static org.junit.Assert.*;
+
+@RunWith(Parameterized.class)
+public class OperationsTest {
```
Эти тесты в зачет темы по тестам?

mpozdeev replied:
Я тесты везде пробую писать.

mpozdeev replied:
Тесты же вроде сданы. Был бы журнал в котором можно было мониторить ситуацию о выполнений заданий (с процентным или числовым показателем выполнения)

mpozdeev committed:
Исправление замечаний

mpozdeev committed:
Added enum, slf4j+logback

Добавлен enum Operations2 содержащий математические операции, приоритет и действие
Логгирование: log4j заменён на slf4j+logback.

mpozdeev committed:
Добавлены изменения

Изменён метод calculate(), функционал по работе с операциями вынесен в отдельный метод operationsRun().
PostfixRecord - удалён неиспользуемый код
